mysql/mysql.go

```diff
@@ -0,0 +1,201 @@
+package mysql
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	// import mysql driver anonymously (just run the init)
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/pkg/errors"
+	nosql "github.com/smallstep/nosql/database"
+)
+
+// DB is a wrapper over *sql.DB,
+type DB struct {
+	db *sql.DB
+}
+
+// Open creates a Driver and connects to the database with the given address
+// and access details.
+func (db *DB) Open(dataSourceName string, opt nosql.Options) error {
+	var err error
+	_db, err := sql.Open("mysql", dataSourceName)
+	if err != nil {
+		return errors.Wrap(err, "error connecting to mysql")
+	}
+	_, err = _db.Exec("CREATE DATABASE IF NOT EXISTS test")
+	if err != nil {
+		return errors.Wrapf(err, "error creating database %s (if not exists)", opt.Database)
+	}
+	db.db, err = sql.Open("mysql", dataSourceName+opt.Database)
+	if err != nil {
+		return errors.Wrapf(err, "error connecting to mysql database")
+	}
+
+	return nil
+}
+
+// Close shutsdown the database driver.
+func (db *DB) Close() error {
+	return errors.WithStack(db.db.Close())
+}
+
+func getQry(bucket []byte) string {
+	return fmt.Sprintf("SELECT nvalue FROM %s WHERE nkey = ?", bucket)
+}
+
+func setQry(bucket []byte) string {
+	return fmt.Sprintf("INSERT INTO %s(nkey, nvalue) VALUES(?,?)", bucket)
+}
+
+func delQry(bucket []byte) string {
+	return fmt.Sprintf("DELETE FROM %s WHERE nkey = ?", bucket)
+}
+
+func createTableQry(bucket []byte) string {
+	//return fmt.Sprintf("CREATE TABLE %s(id int NOT NULL AUTO_INCREMENT, key varchar(255), value BLOB, PRIMARY KEY (key));", bucket)
+	return fmt.Sprintf("CREATE Table %s(id int NOT NULL AUTO_INCREMENT, nkey VARBINARY(255), nvalue BLOB, PRIMARY KEY (id));", bucket)
+
+	//return fmt.Sprintf("CREATE TABLE %s (key varbinary(255), value blob);", bucket)
+}
+
+func deleteTableQry(bucket []byte) string {
+	return fmt.Sprintf("DROP TABLE IF EXISTS %s", bucket)
+}
+
+// Get retrieves the column/row with given key.
+func (db *DB) Get(bucket, key []byte) ([]byte, error) {
+	var val string
+	err := db.db.QueryRow(getQry(bucket), key).Scan(&val)
+	switch {
+	case err == sql.ErrNoRows:
+		return nil, nosql.ErrNotFound
```

Comment:
Taking into account my previous comment on errors, this is a good place to use `errors.Wrapf(nosql.ErrNotFound, "cannot find %s/%s", bucket, key)`